src/eth/rpc/rpc_server.rs

```diff
@@ -273,6 +282,96 @@ fn stratus_reset(_: Params<'_>, ctx: Arc<RpcContext>, _: Extensions) -> Result<J
     Ok(to_json_value(true))
 }
 
+// TODO: improve intermediate steps state
+// TODO: improve error handling
+// TODO: refactor and clean up
+// TODO: add e2e tests
+// TODO: currently leader and follower in eth_sendRawTransaction and stratus_health are defined by consensus being None or not. Need to make it more explicit.
+#[cfg(feature = "dev")]
+async fn stratus_init_importer(params: Params<'_>, ctx: Arc<RpcContext>, _: Extensions) -> Result<JsonValue, StratusError> {
+    if !GlobalState::is_follower() {
+        return Ok(json!(false));
+    }
+
+    if !GlobalState::is_importer_shutdown() {
+        return Ok(json!(false));
+    }
+
+    let app_config: Value = match serde_json::from_value(ctx.app_config.clone()) {
+        Ok(config) => config,
+        Err(_) => {
+            tracing::error!("Failed to parse app_config");
+            return Ok(json!(false));
+        }
+    };
+
+    let importer_config: ImporterConfig = match app_config.get("importer") {
+        Some(importer_value) => match serde_json::from_value(importer_value.clone()) {
+            Ok(config) => config,
+            Err(_) => {
+                tracing::error!("Failed to parse importer configuration");
+                return Ok(json!(false));
+            }
+        },
+        None => {
+            tracing::error!("Importer configuration not found in app_config");
+            return Ok(json!(false));
+        }
+    };
+
+    let (params, external_rpc) = next_rpc_param::<String>(params.sequence())?;
+    let (_, external_rpc_ws) = next_rpc_param::<String>(params)?;
+
+    let importer_config = ImporterConfig {
+        external_rpc,
+        external_rpc_ws: Some(external_rpc_ws),
+        ..importer_config
+    };
+
+    GlobalState::reset_importer_shutdown();
+
+    let consensus = match importer_config
+        .init(Arc::clone(&ctx.executor), Arc::clone(&ctx.miner), Arc::clone(&ctx.storage))
+        .await
+    {
+        Ok(consensus) => consensus,
+        Err(e) => {
+            tracing::error!("Failed to initialize importer: {}", e);
+            return Ok(json!(false));
+        }
+    };
+
+    {
+        let mut consensus_lock = ctx.consensus.write().unwrap();
+        match consensus {
+            Some(consensus) => {
+                *consensus_lock = Some(consensus);
+            }
+            None => {
+                tracing::error!("Failed to update consensus: consensus is None");
+                return Ok(json!(false));
+            }
+        }
+    }
+
+    Ok(json!(true))
+}
+
+#[cfg(feature = "dev")]
+fn stratus_shutdown_importer(_: Params<'_>, _: &RpcContext, _: &Extensions) -> bool {
+    if !GlobalState::is_follower() {
+        return false;
+    }
+
+    if GlobalState::is_importer_shutdown() {
+        return true;
+    }
+
+    const TASK_NAME: &str = "rpc-server::importer-shutdown";
+    GlobalState::shutdown_importer_from(TASK_NAME, "received importer shutdown request");
+    GlobalState::is_importer_shutdown()
```

Comment:
nitpick: this could return true directly.

Reply:
Thanks. I still have on my to do list some code cleanup and small refactor. I will add it to the list

Reply:
Done.